### PR TITLE
feat: setup jsdocs

### PR DESCRIPTION
**Pull request by ZihanLing**
### Description
- Documentation(JSDoc) generation via automation.

### Related Issues
- #17 

### Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Refactor
- [ ] Other (please specify):

### Testing Details
- When commit the changes, the script will add annotation to functions in the source code. It will also automatically generate HTML and CSS files(JSDocs) in automatically created folder named docs. Picture below is an example.
Screenshot 2024-11-25 at 2 19 15 AM

### Checklist
- [x] Code follows the style guide.
- [x] I have performed a self-review of my code.
- [ ] Documentation has been updated (if necessary).
- [x] Tests have been added or updated.

### CODE REVIEW

**Review by kenny-lfx — APPROVED** [review]

File: .husky/pre-commit [hunk 1]
```diff
@@ -1 +1,5 @@
-npx lint-staged
\ No newline at end of file
+npx lint-staged
+npx eslint .
```
Comment (MitchelHsu) [inline .husky/pre-commit:2]: Hmm, why do we have to add `npx eslint .` here? isn't it the same with `npx lint-staged`?

File: scripts/generateJSDoc.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const fs = require('fs');
```
[the remark this answers is not in the document]
Comment (ZihanLing) [reply scripts/generateJSDoc.js:1]: No, you can just change the file path in the script to run the JSDoc. I'll work on finding out if any other features are needed in JSDocs.

File: package.json [hunk 3]
```diff
@@ -1,25 +1,35 @@
...
     "*.{js,mjs}": "eslint --fix",
     "*.css": "stylelint --fix"
   },
   "scripts": {
-    "prepare": "husky"
+    "prepare": "husky", 
+    "generate-docs": "node scripts/generateJsDoc.js", 
+    "docs": "jsdoc -c jsdoc.json"
+  },
+  "husky": {
```
[the remark this answers is not in the document]
Comment (ZihanLing) [reply package.json]: Fix that. Thanks for your advice!

File: .husky/pre-commit (hunk 1 above)
Comment (ZihanLing) [reply .husky/pre-commit]: Fix it. Thanks!

**Review by Shyam-Sundar-Bharathi — APPROVED** [review]

**Pull request merged**